src/core/demangle.d

```diff
@@ -1990,7 +1991,11 @@ pure @safe:
             }
             catch( Exception e )
             {
-                assert( false ); // no other exceptions thrown
+                //assert( false ); // no other exceptions thrown
+                // TODO: ugly hack around mangling error
+                dst = null;
+                dst ~= prevBuf;
+                return dst;
```

Comment:
I still hate this, but I haven't found a better workaround so far (and this is better than just crashing ...).

Reply:
Could you elaborate what exception is thrown that might be handled here?

Reply:
It seems the ParseException isn't caught correctly inside a when demangling "_Dmain". As exceptions are also used for back tracking inside the demangler for valid symbols, it would be nice to actually find the root cause. Maybe _d_throwdwarf doesn't support recursive calls?

Reply:
I think we don't have to block merging this PR on finding the root cause, but you should use the same code as for the ParseException to avoid calling into the GC most of the time, or just catch Exception there instead and add an appropriate comment.

Reply:
FYI I think @ibuclaw managed to find the root cause: https://github.com/dlang/druntime/pull/2245
I removed the mangling workaround and cherry-picked his commit to see whether the CIs agree.